File: Typescript/src/terminal.ts
export const gotoXY = (x: number, y: number) => {
    process.stdout.write(`\x1B[${y};${x}H`);
};

export const hideCursor = () => {
    process.stdout.write("\x1B[?25l");
};

export const showCursor = () => {
    process.stdout.write("\x1B[?25h");
};

export const clearDisplay = () => {
    process.stdout.write("\x1B[2J\x1B[H");
};

File: lua/terminal.lua
local terminal = {}

function terminal.goto_xy(x, y)
    io.write("\x1B[" .. y .. ";" .. x .. "H")
    io.flush()
end

function terminal.hide_cursor()
    io.write("\x1B[?25l")
    io.flush()
end

function terminal.show_cursor()
    io.write("\x1B[?25h")
    io.flush()
end

function terminal.clear_display()
    io.write("\x1B[2J\x1B[H")
    io.flush()
end

return terminal
